### PR TITLE
fix: select only testing files for migration

### PR DESCRIPTION
**Pull request by estrada9166**
- Closes  [UNIFY-1224](https://cypress-io.atlassian.net/jira/software/projects/UNIFY/boards/20?assignee=5ea883deddd45f0b94f5d98d&selectedIssue=UNIFY-1224) [UNIFY-1225](https://cypress-io.atlassian.net/jira/software/projects/UNIFY/boards/20?assignee=5ea883deddd45f0b94f5d98d&selectedIssue=UNIFY-1225)

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/data-context/src/sources/migration/shouldShowSteps.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export function isDefaultTestFiles (config: OldCypressConfig, type: 'component'
     ? getComponentTestFilesGlobs(config)
     : getIntegrationTestFilesGlobs(config)
 
-  return testFiles.length === 1 && testFiles[0] === '**/*'
+  return testFiles.length === 1 && testFiles[0] === '**/*.{js,ts,jsx,tsx,coffee}'
```
Comment (lmiller1990) [inline packages/data-context/src/sources/migration/shouldShowSteps.ts:33]: I'm so confused how this is working at all! I thought that `isDefaultTestFiles` would only be true if the testFiles was `**/*`, which I believe is the 9.x default. I'm quite surprised this didn't break everything. I thought this was used to determine if the user, when migration, has a custom `testFiles` or not.

Maybe this is just working because you changed https://github.com/cypress-io/cypress/pull/20454/files#diff-c2c1f114b54072e7f036c272e41454586ddba1378ada7bf61d38049e05e3d8f5R17 to match?

Should this be a constant perhaps, like `defaultMigrationSpecPattern = '**/*.{js,ts,jsx,tsx,coffee}'`?

**Review by lmiller1990 — COMMENTED** [review]
If it works it works, surprised this doesn't break anything. 

In 9.x we must be excluding `json` files from `cypress/component` and `cypress/integration`? The original bug report in Jira says the problem was a json file in `cypress/component`/

**Review by lmiller1990 — APPROVED** [review]
I was a bit surprised it's working but I think it actually makes sense now - if the tests are all ✅  I say 🚢

File: packages/data-context/src/sources/migration/autoRename.ts [hunk 2]
```diff
@@ -115,7 +115,7 @@ export async function getSpecs (projectRoot: string, config: OldCypressConfig):
   const globs = integrationFolder === false
     ? []
     : integrationFolder === 'cypress/integration'
-      ? ['**/*'].map((glob) => `${integrationFolder}/${glob}`)
```
Comment (emilyrohrbough) [inline packages/data-context/src/sources/migration/autoRename.ts:118]: why is this necessary? Wont' the integrationTestFiles be sufficient since it returns [`['**/*.{js,ts,jsx,tsx,coffee}']`](https://github.com/cypress-io/cypress/blob/alejandro/fix/select-only-testing-files-for-migration/packages/data-context/src/sources/migration/shouldShowSteps.ts#L17) by default?

**Comment by lmiller1990** [comment]
We good to merge this?

**Review by emilyrohrbough — APPROVED** [review]
